### PR TITLE
Prevents /mnt/models/<model name> from being converted into a file

### PR DESCRIPTION
**Pull request by abchoo**
**What this PR does / why we need it**: The model directory will be converted into a file whenever the fileName (Model dir, model name, objectName) is equal to the model directory. This will prevent that override from happening.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by abchoo** [comment]
The GCS test was writing the model content into `/mnt/models/model1` which should be a directory but became a file. Updated this to write to `/mnt/models/model1/testModel1` to reflect the expected behaviour.

File: pkg/agent/storage/gcs.go [hunk 1]
```diff
@@ -70,9 +71,12 @@ func (g *GCSObjectDownloader) Download(client stiface.Client, it stiface.ObjectI
 		if err != nil {
 			return fmt.Errorf("an error occurred while iterating: %v", err)
 		}
-		foundObject = true
 		objectValue := strings.TrimPrefix(attrs.Name, g.Item)
 		fileName := filepath.Join(g.ModelDir, g.ModelName, objectValue)
+		if fileName == filePath {
```
Comment (yuzisun) [inline pkg/agent/storage/gcs.go:76]: So the case here is that objectValue is empty string?

Comment (abchoo) [reply pkg/agent/storage/gcs.go:76]: This case covers when objectValue is `` or `/`.

**Pull request merged**